api-models/course-feedback.yaml

```diff
@@ -0,0 +1,158 @@
+openapi: "3.0.0"
+info:
+  title: Course Feedback API
+  version: "1.0.0"
+  description: API for managing Course feedback with field selection and full data retrieval.
+
+paths:
+  /course-feedback:
+    get:
+      summary: Retrieve a list of all Course feedbacks
+      operationId: listCourseFeedback
+      parameters:
+        - name: fields
+          in: query
+          description: Comma-separated list of fields to include in the response
+          schema:
+            type: string
+        - name: full
+          in: query
+          description: Flag to include nested data in the response
+          schema:
+            type: boolean
+      responses:
+        "200":
+          description: A list of all Course feedbacks
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/schemas/Feedback"
+              examples:
+                fullData:
+                  summary: Response with full=true
+                  value: [{"userId": "762b1737-fee6-4306-bab4-ebed7b5659ed", "courseId": "c3ed8a9e-ac75-4b24-807c-3624c38b49e0", "courseQuality": 5, "courseLength": 5, "nestedData": {"detail1": "Detail 1.1", "detail2": "Detail 1.2"}}, {"userId": "762b1737-fee6-4306-bab4-ebed7b5659ed", "courseId": "762b1737-fee6-4306-bab4-ebed7b5659ed", "courseQuality": 4, "courseLength": 3, "nestedData": {"detail1": "Detail 2.1", "detail2": "Detail 2.2"}}]
+                lengthAndQuality:
+                  summary: Response with fields=courseLength,courseQuality
+                  value: [{"courseLength": 3, "courseQuality": 4}, {"courseLength": 6, "courseQuality": 3}]
```

Comment:
Change according to the comment above

Reply:
Changed the example to fit new only parameter